main/tasks/ping_task.cpp

```diff
@@ -26,6 +28,38 @@ struct PingResult {
     double max_rtt_ms;
 };
 
+struct PingHistory {
+    uint16_t sent;
+    uint16_t received;
+};
+
+static PingHistory ping_history[HISTORY_SIZE];  // buffer storing ping results
```

Comment:
This should consume between 3.6 and 7.2kB of RAM (I think ESP32 is padding the uint16_t to uint32_t without packed attribute). Could you please use the `MALLOC` macro for allocating the space? (it's in `macros.h`) and it puts the memory into the PSRAM, so no previous internal RAM is needed. 

something like:
```c
#include "macros.h"

static struct PingHistory *ping_history = NULL;  // pointer to allocated buffer

#define HISTORY_SIZE 900

// call once at startup
static int init_ping_history(size_t size) {
    ping_history = (struct PingHistory *)MALLOC(size * sizeof(struct PingHistory));
    if (!ping_history) {
        return -1; // allocation failed
    }
    history_index = 0;
    history_count = 0;
    return 0;
}

```
The rest shouldn't need any changes.